webapp/components/survey/SurveyForm/SurveyForm.js

```diff
@@ -75,6 +75,14 @@ const SurveyForm = (props) => {
     }
   }, [])
 
+  useEffect(() => {
+    return () => {
+      if (entry && !preview) {
+        dispatch(SurveyFormActions.resetForm())
```

Comment:
do we need to reset the form in both surveyForm and record?

Reply:
States:
A - Form Designer: `{entry: false, preview: false, edit: true}`
B - Form Designer-preview: `{entry: true, preview: true edit: false}`
C- Entry record: `{entry: true, preview: false edit: false}`

Goal: (After closing preview, in form designer show the last entity shown in preview)

Cases: 
A - B -> reset ( not reset if we like to move from designer to preview and keep the page)
B - A -> not reset -> fix
A - C -> reset
B - C -> reset
C - X -> reset

Reply:
B->C and C->B do not exists .

cant you reset the form only when you unmount the surveyForm?

Reply:
B-C is possible, go to Designer, open Preview go-to records
C-B agree

If we only reset on unmount of SurveyForm is going to reset always: SurveyFrom is used on the three states.